app/views/admin/users/_activity.html.erb

```diff
@@ -15,7 +15,7 @@
     <p>(view full history in notes below)</p>
     <ul>
       <% @user.roles.each do |role| %>
-        <li><%= role.name %> <em><%= "- #{Tag.find(role.resource_id).name}" if role.resource_id.present? %></em></li>
+        <li><%= role.name %> <em><%= role.resource_name.to_s %></em></li>
```

Comment:
_**Quick, non-blocking thought**_: I could foresee a scenario in which someone is assigned to be a tag mod for a handful of tags (10+) in addition to any roles that they currently have, making this roles section a bit long. My local user has 3 admin-related roles and is a tag mod for 5 different tags and the list seems not too far from the "bit too long" scenario. This being said, it might be nice to paginate this section (or have a roles dropdown -- something that doesn't hide the role history but doesn't make the page super long) at some point.

Reply:
For sure, I'll wait for the designers to get to this page before I attempt anything though :laughing: